pkgs/development/libraries/pipewire/default.nix

```
@@ -28,6 +27,8 @@
 , makeFontsConf
 , callPackage
 , nixosTests
+, withValgrind ? lib.meta.availableOn stdenv.hostPlatform valgrind
+, valgrind ? null
```

I've seen multiple mentions that `? null` should not be used as parameters are lazily evaluated anyway and adding this fallback can hide other problems down the line.

---

This masks for example if an undefined variable is used and this is not required for this to function properly. valgrind is also defined on darwin and it theoretical it must also eval on all platforms but since nix is lazy, how @jansol already wrote, it is not even evaled beyond the point that the package exists.

```suggestion
, valgrind
```

---

Should I also remove all `? null` below? Seems there are a lot.

---

Please do. Might as well take care of them all in one go so that is off the table and further attention can go where it is needed more.